### DISCUSSION TITLE
Running beautifully

### DISCUSSION THREAD

**Discussion by bobwolferocks** [JPyke3/mbp-manjaro#15] [Q&A]
Thanks for this man. It works really well and the installation instructions were super clear.
Just got some issues when updating software or downloading and installing packages:
error: failed retrieving file 'mbp.db' from mbp-repo.jacobpyke.xyz : The requested URL returned error: 530
I've made sure to use your latest repository. I'm in China so having issues installing a VPN that works. Can't use astrill on Manjaro.
This might be the cause of my problems. If I can just get a VPN to work here that would be great then I can support you on Patreon.
Thanks!

**Comment by JPyke3** [comment]
Hey there! Thanks for posting in the discussions! This doesn’t actually appear to be an error on your side at all! 

The file hosting Provider I use is called "Wasabi". And it seems like they Are having problems with their DNS Lookups.

Here is their status now:
https://status.wasabi.com/pages/5abbc12aeb57a904e44a58d3

I won’t lie that’s really surprising that they are having these issues. They are usually known for high reliability. If the issue persists too long, I have everything stored locally so I can setup another mirror.

**Reply by JPyke3** [reply]
Keep in mind that the region that our files are on is `eu-central-1` as it’s geographically the closest to me.

**Reply by JPyke3** [reply]
Oh, it seems like they Are having issues with their domain, they maybe failed to renew it or lost control of it? Here is their incident report:

We are currently awaiting the restoral of the wasabisys.com domain by our third-party domain name provider.  We do not have a restoral ETA at this point because we are dependent on the third-party domain name provider to take actions.

**Reply by bobwolferocks** [reply]
thanks, man. I'll wait a while and try again. I'm pretty new to Linux so it's lots to learn. I tried Linux mint, Ubuntu and now using Manjaro as this was the most clear and comprehensive solution for linux on a recent mbp. Going to try to get wifi working soon.
THanks!

**Reply by JPyke3** [reply]
No stress at all! I’ll be monitoring everything over the day today and will write here when I see that wasabi is up again. Luckily if you’re getting a 530 error at all it means you are at least hitting Cloudflare which is my CDN. Which I would therefore assume means your VPN solution whatever it is, is working. 

Good luck with the WiFi! Feel free to post here if you have any questions!

**Reply by bobwolferocks** [reply]
Great! I actually think it will connect without a VPN. I don't know if I can get astrill working or not. Still working on it. Here is what happened when I tried to install it:

Installing reconnect service
/usr/local/Astrill/astrill: error while loading shared libraries: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory
WARNING: Your system is missing pixbuf theme engine. Please install package gtk2-engines-pixbuf or gtk2-engines
Creating icons...
Installing LSP...
Testing '/usr/local/Astrill/liblsp64.so':PASSED

Any ideas?

**Reply by JPyke3** [reply]
Yep, looks like your missing a library. How did you install it? Via the AUR? In any case apparently it doesn’t install `gtk2`. You can get around this by just installing it:
```
sudo pacman -S gtk2
```

**Reply by bobwolferocks** [reply]
cool, thanks! I'll give it a go when the domain is up and running again.
I just installed in via terminal after downloading the .sh file from the astrill site.

**Reply by bobwolferocks** [reply]
It's all back up and running now. Doing a whole load of updates. Should be able to get a VPN sorted soon.
Cheers!

**Reply by JPyke3** [reply]
Oh wow that it is, i checked no joke 5 minutes before you posted this and it wasn’t up for me. Oh well glad it’s up and working now for you! Lmk how you get on with your VPN or if you need a hand :)

**Reply by bobwolferocks** [reply]
VPN is all sorted. Looking at WIFI now. Bit confused. My files are of the C-4364 s-B3 variety. I click on the corresponding files for that type but don't get what I'm supposed to do there. Doesn't seem clear to me sorry.

**Reply by bobwolferocks** [reply]
As well as no wifi yet I have a few issues. Generally it's working well and has been a good experience.
The biggest issue is that it shuts down or becomes unresponsive after leaving for a while. Left it doing a timeshift snapshot before going to bed and woke up to it turned off and couldn't get it to turn on unless I unplugged the power cord and plugged it back in.

I still can't get the built in speakers or the headphone jack to work. I can get my USB Focusrite sound card working with all apps so that's good though.

Keyboard layout seems to be different in the login screen after reboot where " and @ are switched around. Once logged in it's fine and is also fine if I log out. Only happens after rebooting.

The keyboard backlight stays on all the time even when screen is off. Then it will turn off after a while randomly.

Battery life is only about 2-3 hours. Any apps or settings to improve this?

CPU temps seems ok. Runs a little hotter gaming than it did on bootcamp windows. 
Any apps or configurations that you can recommend to optimise the system. 

I know this is a work in progress and it can only get better I'm sure so thanks for everything so far. Next payday I'll be starting Patreon support. Would love to keep using Manjaro as my main OS so need all the help I can get.

**Reply by JPyke3** [reply]
Sorry only just woken up. In case you wonder I am in CEST. Just in case I don’t reply for ages. I’ll reply in a moment when I’m at my PC.

**Reply by bobwolferocks** [reply]
thanks man, all good. I just connected to discord. Seems like sleep is not working yet and lock screen too. so I've set everything to just turn off screen and not sleep or lock or suspend. Seems like battery life is just as bad on mac os I think.
I installed some things to help with battery life according to this page:  https://wiki.manjaro.org/index.php/Power_Management
I have headphones working just not built in speakers and no wifi but no rush. I think just having sleep issue sorted will keep me happy for now. cheers

**Reply by JPyke3** [reply]
Sweet, I threw a bunch of info in discord. But here it is as well just in case:

> I still can't get the built in speakers or the headphone jack to work. I can get my USB Focusrite sound card working with all apps so that's good though.

Okay so as Redecorating suggested have you put your apple-t2.conf in the other alternative location? /usr/share/alsa-card-profile/mixer/profile-sets/apple-t2.conf. If not what MBP do you have? Is it a 16.1? They have a different speaker config

> Keyboard layout seems to be different in the login screen after reboot where " and @ are switched around. Once logged in it's fine and is also fine if I log out. Only happens after rebooting.

What version of Manjaro are you using? Sadly the displaymanager is different between each version of Manjaro so I am sure I can fix it if you let me know which one it is :)

> The keyboard backlight stays on all the time even when screen is off. Then it will turn off after a while randomly.

I'm surprised your keyboard backlight even turns off at all. Mine just stays on constantly. However this could be related to your next problem:

> The biggest issue is that it shuts down or becomes unresponsive after leaving for a while. Left it doing a timeshift snapshot before going to bed and woke up to it turned off and couldn't get it to turn on unless I unplugged the power cord and plugged it back in.

Which version of the kernel are you running if you run uname -a? is it 5.7.x? Anything other than that is considered pretty unstable at the moment.

> Battery life is only about 2-3 hours. Any apps or settings to improve this?
CPU temps seems ok. Runs a little hotter gaming than it did on bootcamp windows.
Any apps or configurations that you can recommend to optimise the system.

This is kinda a 2 for one here. Because Apple does such aggressive CPU Frequency Scaling in MacOS, none of those configurations are here in Linux. Windows is okaaay because they have some scaling. I recommend looking at this: [thermald](https://wiki.archlinux.org/index.php/CPU_frequency_scaling#thermald). I use thermald personally, I find that it does a really good job of keeeping the temps in order, as well as holding the CPU scaling a bit back to considerably improve battery.

> Bit confused. My files are of the C-4364 s-B3 variety. I click on the corresponding files for that type but don't get what I'm supposed to do there. Doesn't seem clear to me sorry.

Would you be able to post the output of ioreg -l | grep C-4364 from MacOS here in discord chat? I can give you some commands that will just put the files in the correct place.

**Reply by bobwolferocks** [reply]
here is the ioreg output:
 | |   |       | |   "RequestedFiles" = ({"Firmware"="C-4364__s-B3/bali.trx","TxCap"="C-4364__s-B3/bali-X2.txcb","Regulatory"="C-4364__s-B3/bali-X2.clmb","NVRAM"="C-4364__s-B3/P-bali-X2_M-HRPN_V-m__m-7.9.txt"})

    | |   |       | |       "images" = {"C-4364__s-B3/bali-X2.txcb"={"imagetype"="TxCap","required"=No,"imagename"="C-4364__s-B3/bali-X2.txcb"},"C-4364__s-B3/bali-X2.clmb"={"imagetype"="Regulatory","required"=Yes,"imagename"="C-4364__s-B3/bali-X2.clmb"},"C-4364__s-B3/P-bali-X2_M-HRPN_V-m__m-7.9.txt"={"imagetype"="NVRAM","required"=Yes,"imagename"="C-4364__s-B3/P-bali-X2_M-HRPN_V-m__m-7.9.txt"},"C-4364__s-B3/bali.trx"={"imagetype"="Firmware","required"=Yes,"imagename"="C-4364__s-B3/bali.trx"}}

stevesmith@Steves-MacBook-Pro ~ %

**Reply by JPyke3** [reply]
Oh and any chance you can let me know what model your Mac is?

https://support.apple.com/en-us/HT201300

**Reply by bobwolferocks** [reply]
Mine is either 16.1 or 16.4

**Reply by JPyke3** [reply]
Ahhhh, unfortunately those models don't have a supported WiFi chip. After MacOS Mojave, apple switched how it's firmware files are stored so that any MacBook that never had MacOS Mojave has binary incompatible files with Linux :(

In order to get your WiFi chip working we have to reverse engineer the firmware files and recreate them for Linux support. I am pretty sure @aunali1 is working on this. But unfortunately it's a pretty big undertaking. No promises anytime soon. Sorry about this, we are working on it as quick as we can.

**Reply by JPyke3** [reply]
If you absolutely need WiFi a USB WiFI adapter will work. I understand this is quite the compromise however.

**Reply by bobwolferocks** [reply]
ah I see, no worries. Things left to do then is sound and the keyboard input at login. I'm using the KDE version and the kernel is: 5.7.9-2.
I think I have sorted everything else. I've disabled sleep and suspend and I've installed termalD. 
I don't really move around the apartment with it anyway so wifi is not a big deal right now but would be good to get it working eventually.
BTW I've tried a few things for the sound. Namely stuff on this page: https://gist.github.com/kevineinarsson/8e5e92664f97508277fefef1b8015fba

**Reply by JPyke3** [reply]
Okay so Keyboard:

Looks like `sddm` just uses your locale keyboard layout to detect it's keyboard layout. What is your output of `localectl status`?

About the sound:

Have you seen this commend on the gist you linked? https://gist.github.com/kevineinarsson/8e5e92664f97508277fefef1b8015fba#gistcomment-3502937

Seems like there has been a recent pulse update that changes where the files are read from.

**Reply by bobwolferocks** [reply]
Ok login seems to be ok now after I changed the keyboard settings

output of localectl status:
System Locale: LANG=en_GB.UTF-8
                  LC_NUMERIC=aa_ER
                  LC_TIME=aa_ER
                  LC_MONETARY=aa_ER
                  LC_PAPER=aa_ER
                  LC_NAME=aa_ER
                  LC_ADDRESS=aa_ER
                  LC_TELEPHONE=aa_ER
                  LC_MEASUREMENT=aa_ER
                  LC_IDENTIFICATION=aa_ER
       VC Keymap: n/a
      X11 Layout: au
       X11 Model: macbook79

**Reply by bobwolferocks** [reply]
![Screenshot_20210101_091142](https://user-images.githubusercontent.com/74397998/103431848-b0d77f80-4c11-11eb-9e56-c033d6b050e8.png)

**Reply by JPyke3** [reply]
Good to hear you got the keyboard errors fixed! 

As far as the pulseaudio errors go, we probably have to take a look at why pulse isn't loading anything correctly. Would you be able to run:
```
journalctl | grep pulse
```

Then we can take a look at what it's doing, if it's alot of output feel free to dump it all into a pastebin or whatever.

**Reply by JPyke3** [reply]
Because to me your pulse rules look totally fine.

**Reply by bobwolferocks** [reply]
https://pastebin.com/DW314Rr3

**Reply by JPyke3** [reply]
I moved all your output to a pastebin, sorry it was a bit too long

**Reply by JPyke3** [reply]
To me that looks like pulse is launching totally fine, I am not sure if it's configured correctly perhaps in Plasmas settings? What audio devices do you see within KDE's settings?

**Reply by bobwolferocks** [reply]
![Screenshot_20210102_162219](https://user-images.githubusercontent.com/74397998/103453642-df8b4e00-4d16-11eb-8f58-1cd41abcf91d.png)

**Reply by JPyke3** [reply]
Okay, well "Apple Audio Device" is the speakers. So that means the Pulse rules are working and Pulse is intializing correctly. Perhaps switch your "Output" to "Apple Audio Device"

**Reply by bobwolferocks** [reply]
When the scarlett USB card is turned off and built in speakers are selected for there to be an output option i need to select "Add virtual output device for simultaneous output..."
That gives me two output options that are both called simultaneous output. One has front left, front center and front right and the other has those as well as rear left, rear right and subwoofer. 
I don't get an output option if built in speaker is selected and  "Add virtual output device for simultaneous output..." is not selected.

**Reply by bobwolferocks** [reply]
![Screenshot_20210102_172133](https://user-images.githubusercontent.com/74397998/103454475-85db5180-4d1f-11eb-8859-5f6306af149c.png)

**Reply by JPyke3** [reply]
Oh right I see, sorry, I am not 100% familiar with KDE and it's settings, however, I think maybe then you just have to change the output option to `Apple Audio Device` in the `Devices` tab up the top? Essentially you just need to pump all audio to that device and it should output totally fine. Pulse loaded correctly as did the config as you can see the speaker arrangement for testing.

**Reply by bobwolferocks** [reply]
i went through all of the files to make sure they had the same code/text as the files on https://gist.github.com/kevineinarsson/8e5e92664f97508277fefef1b8015fba
and got it to work ok by adding what wes974 suggested: 

"I added every files but I can’t seems to get the sound working on my MBP16.
EDIT:
I got it working by placing the provided default.pa and daemon.conf into /etc/pulse and adding realtime-scheduling = no to daemon.conf"

This got sound the to the point where it works but not very well. The sound is very harsh and tinny and mostly comes out of the left side at the back and doesn't seem like all speakers are being utilised. The speaker system on this MBP is particularly good and sounds amazing for laptop speakers. I wonder how I can get it working like it should.

**Reply by bobwolferocks** [reply]
Maybe the mapping of the speakers need configuring?
While doing the audio output testing in the advanced tab:
-front left works(sounds pretty good)
-rear left works(sounds thin, not very good)
-front right comes out of front left speaker
-rear right comes out of front right speaker
-front centre comes out of front right speaker. 
-sub woofer(not sure why it's there) just has static/white noise. 
I'm going to reboot into mac os to see if I can do a similar test to see what the speakers do and where sound comes from.

**Reply by JPyke3** [reply]
To me that sounds like you have an incorrect config, it sounds like you have the Config file from a 13" MBP or the older 15" whereas the 16 has a different speaker setup, make sure you remove `audio-fix-mbp`
```
sudo pacman -R audio-fix-mbp
```

And perhaps download the files you linked again and put everything in place once more just in case something gets overwritten. Sadly I cannot test as I don't own this machine, your best bet would be on Discord. However, this isn't an issue with the Distro, you can ask this in the general channel as this isn't specific to Manjaro. From my side pulse seems to be initialising correctly. Report back here how you get on as I will update the Wiki accordingly.

**Reply by bobwolferocks** [reply]
ah ok! I did that and at least the left and right are coming from the correct place. Only front speakers are working I think though.
Thanks for your help. I'll ask on discord and see what I can find out.

**Reply by bobwolferocks** [reply]
One last thing. I currently don't have bootcamp windows but if I wanted to do that is there a way I can do it without messing up the bootloader? Also, how can I boot automatically into Manjaro instead of mac os? 
Thanks

**Reply by JPyke3** [reply]
In terms of installinig Windows. There aren't actually any files that you have installed that should impact the windows bootloader. *However* I am not sure if installing windows will clean the EFI partition just as default. Worst comes to worse. You can try installing windows. if it messes up your EFI partition, you can boot the liveusb again and reinstall the bootloader. You can do that by following this: https://github.com/JPyke3/mbp-manjaro/blob/beta-0.1/README.md from step 16 down.

**Reply by bobwolferocks** [reply]
awesome, no problems booting to all 3 OSes now. thanks a lot!

**Reply by JPyke3** [reply]
Good to hear!

Feel free to reply here agian if you need anything else or on Discord! :D

**Reply by bobwolferocks** [reply]
Hi, mate. How are things? Manjaro still going well for me. I sent you a message on patreon about a wifi adapter but not sure you got the notification. Anyway, looks like aunali1 is close to getting wifi working on the 16,1? 
Just wondering which version of Manjaro you're running and why you chose that? I'm happy with KDE but there are a few quirks here and there. One is that when viewing battery percentage it no longer shows time remaining. It used to be consistent and accurate then became a bit sporadic and now doesn't show anything.

**Reply by JPyke3** [reply]
Hey mate, good thanks. Sorry about the late response. The past few months have just been totally hectic for me as I started full time work. So I am sorry about the radio silence. Yes I am keeping a close eye on the progress related to WiFi. I am very well aware that this is our biggest request because there seem to be a large portion of machines that do not have it yet. I will push those patches as soon as I get them :)

I think my base image was actually GNOME. but I have kind of hacked it apart ever since. I have completely removed GNOME and actually use i3wm as my Desktop Environment. I tend to find it very useful for lots of programming sessions which I do alot.

Battery percentage is interesting. My battery percentage seems to work just fine. I have cinnamon also installed so I checked there. It always shows correctly. But will max out at 84%. That's because the System only reports the battery percentage in relation to the battery capacity. So 84% is the current amount my battery can hold so that's what is represented

**Comment by gellatofresh** [comment]
Heya m8, doing well 

Gotta get back on GitHub, should be secured i recon. Lol just got two yubikey.

I’m running the latest, the primary reason is mainly for security hardening. 

that’s odd about the battery and definitely annoying though small! I’ll check into that too.

**Comment by gellatofresh** [comment]
Ahh, what kinda machine ya running?

**Reply by JPyke3** [reply]
My MBP is a 2018 13" with 4 Thunderbolt Ports

**Comment by gellatofresh** [comment]
Same model here bro!

**Reply by bobwolferocks** [reply]
Hey,
I attempted to get wifi working.
I briefly tried this Kernel https://github.com/jamlam/mbp-16.1-linux-wifi and followed your wifi instructions but couldn't figure it out. Also, with that Kernel it messed up the touchbar. 
Anyway I guess I'l wait for your new kernel, at least I know how to choose the new kernel to boot from in manjaro.conf file.... learning slowly!
Cheers!
